### PR TITLE
OD-571 [Feature] Added translation to Widget-FormBuilder

### PR DESCRIPTION
**Pull request by OleksandrLavrynenko**
@sofiiakvasnevska @inna-bieshulia

OD-571 https://weboo.atlassian.net/browse/OD-571

## Description
Added user translations in all FormBuilder components

## Backward compatibility
This change is fully backward compatible.

## Reviewers
@upplabs-alex-levchenko

### CODE REVIEW

File: js/components/date.js [hunk 1]
```diff
@@ -53,7 +51,7 @@ Fliplet.FormBuilder.field('date', {
 
     if (Fliplet.Env.get('platform') === 'web') {
       this.datePicker = $(this.$el).find('input.date-picker').datepicker({
-        format: 'yyyy-mm-dd',
+        format: 'yyyy/mm/dd',
```
Comment (squallstar) [inline js/components/date.js:54]: Why this format?

File: translation.json [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            "dataSource": {
+                "loading": "Loading...",
+                "sending": {
+                    "requiredFields": "* Required fields"
+                },
+                "email": {
+                    "invalid": "The input is not a valid email address."
+                },
+                "file": {
+                    "uploaded": "Uploaded:",
```
Comment (squallstar) [inline translation.json:13]: I think the colon should not be part of the translation string

File: translation.json [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                },
+                "radio": {
+                    "defaultOptions": {
+                        "option1": "Option 1",
+                        "option2": "Option 2"
+                    }
+                },
+                "select": {
+                    "name": "Dropdown (single-select)",
+                    "placeholder": "-- Select one",
```
Comment (squallstar) [inline translation.json:60]: The two dashes should not be part of the string

Comment (tonytlwu) [reply translation.json:60]: Agreed

File: translation.json (hunk 2 above)
Comment (tonytlwu) [reply translation.json:13]: Agreed

File: translation.json [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                    "defaultOptions": {
+                        "option1": "Option 1",
+                        "option2": "Option 2"
+                    }
+                },
+                "telephone": {
+                    "instruction": "Phone could contain <b>; , . ( ) - + SPACE * #</b> and numbers."
```
[the remark this answers is not in the document]
Comment (tonytlwu) [reply translation.json:67]: Let's change the instruction to `Phone number can only contain <b>; , . ( ) - + SPACE * #</b> and numbers.`.

I think we should be able to set the translation in JSON this way. However, just as we had done for https://github.com/Fliplet/fliplet-widget-lock/pull/25#discussion_r697604212 it means that translation string must be applied as HTML.

File: templates/components/telephone.build.hbs [hunk 5]
```diff
@@ -24,5 +24,5 @@
   :placeholder="placeholder"
   tabindex="0"
 />
-<p class="text-danger" v-if="$v.value.required === false && $v.value.$dirty">Field is required.</p>
-<p class="text-danger" v-if="$v.value.phone === false && $v.value.$dirty">Phone could contain <b>; , . ( ) - + SPACE * #</b> and numbers.</p>
+<p class="text-danger" v-if="$v.value.required === false && $v.value.$dirty">{{T "widgets.formBuilder.dataSource.errors.required"}}</p>
+<p class="text-danger" v-if="$v.value.phone === false && $v.value.$dirty">{{T "widgets.formBuilder.dataSource.telephone.instruction"}}</p>
```
Comment (tonytlwu) [inline templates/components/telephone.build.hbs:28]: Please share a screenshot confirming that `{{T "widgets.formBuilder.dataSource.telephone.instruction"}}` works.

Comment (OleksandrLavrynenko) [reply templates/components/telephone.build.hbs:28]: > Пожалуйста, поделитесь снимком экрана, подтверждающим, что это `{{T "widgets.formBuilder.dataSource.telephone.instruction"}}`работает.

```suggestion
<p class="text-danger" v-if="$v.value.phone === false && $v.value.$dirty">{{T "widgets.formBuilder.dataSource.telephone.instruction"}}</p>
![tel-error](https://user-images.githubusercontent.com/86256215/134631899-10431e73-e50b-460e-b8ee-afccda7b588a.jpg)

```

Comment (OleksandrLavrynenko) [reply templates/components/telephone.build.hbs:28]: > Пожалуйста, поделитесь снимком экрана, подтверждающим, что это `{{T "widgets.formBuilder.dataSource.telephone.instruction"}}`работает.
tel-error

File: js/components/date.js (hunk 1 above)
Comment (OleksandrLavrynenko) [reply js/components/date.js:54]: > Why this format?

I used the function TD(date, {format: 'L'}) to translate the date. This function returns date in format 'yyyy/mm/dd'. So I had to change the format of DataPicker.

File: .eslintrc [hunk 6]
```diff
@@ -3,6 +3,9 @@
   "globals": {
     "Modernizr": "readonly",
     "Vue": "readonly",
-    "Sortable": "readonly"
+    "Sortable": "readonly",
+    "TD": "readonly",
```
Comment (tonytlwu) [inline .eslintrc:7]:
```suggestion
```

File: translation.json [hunk 7]
```diff
@@ -0,0 +1,83 @@
+{
+    "widgets": {
+        "formBuilder": {
+            "dataSource": {
```
Comment (tonytlwu) [inline translation.json:4]:
```suggestion
```

File: translation.json [hunk 8]
```diff
@@ -0,0 +1,83 @@
+{
+    "widgets": {
+        "formBuilder": {
```
Comment (tonytlwu) [inline translation.json:3]:
```suggestion
        "form": {
```

File: translation.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (tonytlwu) [reply translation.json:67]: Pending action

File: templates/components/telephone.build.hbs (hunk 5 above)
Comment (tonytlwu) [reply templates/components/telephone.build.hbs:28]: 👍🏻

File: js/components/date.js (hunk 1 above)
Comment (tonytlwu) [reply js/components/date.js:54]: Please do not include any date/number localisation in this work. This PR should only include work on translations.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: translation.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (tonytlwu) [reply translation.json]: Pending action

**Review by tonytlwu — CHANGES_REQUESTED** [review]
Please address all unresolved conversations.

**Comment by squallstar** [comment]
FYI this PR has conflicts to be fixed.

**Comment by squallstar** [comment]
@OleksandrLavrynenko can you make sure this branch is up to date with master? That ensures no conflicts are raised for this PR since it has not been merged yet but eventually will.

**Comment by OleksandrLavrynenko** [comment]
> @OleksandrLavrynenko can you make sure this branch is up to date with master? That ensures no conflicts are raised for this PR since it has not been merged yet but eventually will.

@squallstar yes, this branch is up to date with master

**Review by tonytlwu — APPROVED** [review]

**Pull request merged**